react-piper/src/utils/serialize.js

```diff
@@ -96,7 +96,7 @@ const genInputByHandle = (nodes, edges, targetHandle) => {
 
     let path = (srcNode.data.type === 'input' || srcNode.data.type === 'map_input') ? 'input' : srcNode.id;
     if (srcExtraOutput.length !== 0) {
-        const name = srcExtraOutput[0].name;
+        const name = srcExtraOutput[0].name.replace(/\[i]/g, `.${srcExtraOutput[0].index || 0}`);
```

Comment:
there is an error. cases like ".coords[i].qwe[i]" will lead to incorrect results.